### PR TITLE
Unalign fix rebased

### PR DESCRIPTION
**Pull request by monicasee**

### CODE REVIEW

**Comment by monicasee** [comment]
test:jenkins/ch3/all

**Comment by monicasee** [comment]
test:jenkins/ch3/all

**Comment by monicasee** [comment]
test:jenkins/ch4/all

**Comment by monicasee** [comment]
test:jenkins/ch3/tcp

**Comment by monicasee** [comment]
test:jenkins/ch3/sock

**Comment by monicasee** [comment]
test:jenkins/ch3/all

File: src/mpl/include/mpl_trmem.h [hunk 1]
```diff
@@ -113,6 +113,78 @@ char *MPL_strdup(const char *str);
 #endif /* MPL_USE_MEMORY_TRACING */
 
 
+/* --------------------------------------------------------------------
+ * MPL memory alignment union
+ * The MPL_mem_alignment_t union is used to help internal structures or buffers
+ * follow the alignment rules for all predefined datatypes. */
+
+#if 0 /* sample usage : wrap up sample_t structure. */
```
Comment (pavanbalaji) [inline src/mpl/include/mpl_trmem.h]: What's the point of this commented out code?

File: src/mpl/include/mpl_trmem.h [hunk 2]
```diff
@@ -113,6 +113,78 @@ char *MPL_strdup(const char *str);
...
+    sample_t var;
+    MPL_mem_alignment_t alignment;
+} aligned_sample_t;
+#endif
+
+typedef struct {
+    double re;
+    double im;
+} mpl_d_complex;
+
```
Comment (pavanbalaji) [inline src/mpl/include/mpl_trmem.h]: You don't need a separate named structure here. You can simply declare the structure inside the union you are using it in.

File: src/mpl/include/mpl_trmem.h [hunk 3]
```diff
@@ -113,6 +113,78 @@ char *MPL_strdup(const char *str);
...
+#ifdef HAVE_LONG_DOUBLE
+typedef struct {
+    long double re;
+    long double im;
+} mpl_ld_complex;
+#endif
```
Comment (pavanbalaji) [inline src/mpl/include/mpl_trmem.h]: Same comment as above.

File: src/mpl/include/mpl_trmem.h (hunk 1 above)
Comment (monicasee) [inline src/mpl/include/mpl_trmem.h]: It is a sample code, showing how to use this alignment structure.

**Comment by monicasee** [comment]
test:jenkins/ch3/tcp

**Pull request merged**